From 430b49a17642d54182d1ce7451c4a6a9141f0340 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Tue, 31 Mar 2020 22:50:43 +0300
Subject: [PATCH 8/8] meson: Pass win32_ldflags as arguments to test
 executables

This makes sure to link the test executables against ole32.dll.

Normally Meson links by default links against a few common windows
libraries, including ole32.dll, via the c_winlibs property. However,
Meson missed doing that for Clang in mingw mode, which was fixed
upstream in Meson in https://github.com/mesonbuild/meson/pull/6727,
part of the Meson 0.54.0 release.

This fixes linking the tests with Clang on mingw with older Meson
releases. It's not sent upstream, as it's only necessary for building
with older Meson versions.
---
 gio/tests/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gio/tests/meson.build b/gio/tests/meson.build
index 788cf978b..295691485 100644
--- a/gio/tests/meson.build
+++ b/gio/tests/meson.build
@@ -755,6 +755,7 @@ foreach test_name, extra_args : gio_tests
   exe = executable(test_name, [source, extra_sources],
     c_args : test_c_args + extra_args.get('c_args', []),
     dependencies : common_gio_tests_deps + extra_args.get('dependencies', []),
+    link_args: [win32_ldflags],
     install_rpath : extra_args.get('install_rpath', ''),
     install_dir: installed_tests_execdir,
     install: install,
-- 
2.17.1

